### PR TITLE
add length property for all cheerio instances

### LINKED ISSUE #1674: .merge() - wont accept Cheerio object.
### Test

``` js
const cheerio = require('cheerio');

const ret = cheerio.merge(new cheerio(), ["elem1","elem2"]);
```
Problem is **merge** uses function **isArrayLike** and **isArrayLike** uses **Object.hasOwnProperty** for testing does object has **length** property. Cheerio objects has **length** property, but it is defined in prototype and it wont meet criteria.

### Option 1

Remove check and let it throw if object has not array like

### Option 2

maybe test is property length a number instead

### PR DESCRIPTION
**Pull request by 5saviahv**
* Add length parameter for all cheerio instances.
* re do merge with simple postfix operators

should resolve #1674

### CODE REVIEW

File: lib/cheerio.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ var Cheerio = (module.exports = function (selector, context, root, options) {
     return new Cheerio(selector, context, root, options);
   }
 
+  this.length = 0;
```
Comment (fb55) [inline lib/cheerio.js:43]: Remove the prototype property from L109?

Comment (5saviahv) [reply lib/cheerio.js:43]: Done

**Comment by fb55** [comment]
This looks great! I should have flagged this earlier — could you include a test (probably based on #1674) to make sure this doesn't break again?

**Comment by fb55** [comment]
Awesome, thanks a lot @5saviahv!

**Pull request merged**